From f0a1dd1e764febf5527f6ff94a138b41c0b2f0c9 Mon Sep 17 00:00:00 2001
From: Martin Kosek <mkosek@redhat.com>
Date: Wed, 2 Jan 2013 10:43:49 +0100
Subject: [PATCH] Do not crash when Kerberos SRV record is not found

ipa-client-install crashed when IPA server realm TXT record was
configured, but the referred domain (lower-case realm value) did
not contain any Kerberos SRV record (_kerberos._udp.<realm>.)

https://fedorahosted.org/freeipa/ticket/3316
---
 ipa-client/ipaclient/ipadiscovery.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/ipa-client/ipaclient/ipadiscovery.py b/ipa-client/ipaclient/ipadiscovery.py
index e20cd64ec51153eea086afdeb6a1f05775bd72b3..18b77a684505a61b4958cde678fc58b66c6632fd 100644
--- a/ipa-client/ipaclient/ipadiscovery.py
+++ b/ipa-client/ipaclient/ipadiscovery.py
@@ -426,9 +426,10 @@ class IPADiscovery(object):
             kdc = self.ipadns_search_srv(domain, '_kerberos._udp', 88,
                     break_on_first=False)
 
-            if not kdc:
+            if kdc:
+                kdc = ','.join(kdc)
+            else:
                 root_logger.debug("SRV record for KDC not found! Realm: %s, SRV record: %s" % (realm, qname))
                 kdc = None
-            kdc = ','.join(kdc)
 
         return realm, kdc
-- 
1.7.11.7

